src/routes/profile-route/profile-settings-route.jsx

```diff
@@ -0,0 +1,205 @@
+import { useMutation, useQuery } from "@apollo/client";
+import { useContext, useEffect } from "react";
+import { useForm } from "react-hook-form";
+import { useMediaQuery } from "react-responsive";
+import { DeleteUserModal } from "../../components/delete-user-modal";
+import { Footer, Form, Header } from "../../components/index";
+import { GET_USER, GET_USERS, UPDATE_USER } from "../../graphql/";
+import { useModals } from "../../store";
+import { CurrentUserContext } from "../../store/current-user-context";
+import { ProfileHorizontalMenu, ProfileSidebar } from "./index";
+
+export const ProfileSettingsRoute = () => {
+  const { currentUser, setCurrentUser } = useContext(CurrentUserContext);
+  const userId = currentUser.id;
+  const { registerModal, openModal, closeModal } = useModals();
+
+  const prefilledInputs = {
+    username: currentUser?.username,
+    email: currentUser?.email,
+  };
+
+  const userQuery = useQuery(GET_USER, {
+    variables: {
+      where: {
+        id: userId,
+      },
+    },
+  });
+
+  const usersQuery = useQuery(GET_USERS, { variables: { where: {} } });
+
+  const [updateUser, { data, loading, error }] = useMutation(UPDATE_USER, {
+    refetchQueries: [GET_USER],
+  });
+
+  // - - - - - - - - - - USE EFFECT - - - - - - - - - -
+  useEffect(() => {
+    registerModal("deleteUser", <DeleteUserModal onClose={closeModal} />);
+  }, []);
+
+  // - - - - - - - - - - USE FORM - - - - - - - - - -
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isValid },
+    setValue,
+    getValues,
+  } = useForm({
+    defaultValues: {
+      username: "",
+      email: "",
+      password: "",
+    },
+  });
+
+  const onSubmit = () => {
+    //loading should be true
+    //get values from form
+    const { username, email, password } = getValues();
+    //make sure that new username is unique
+    //const usersQuery = useQuery(GET_USERS, { variables: { where: {} } });
+    //usersQuery.map((username) => {
+    //});
+
+    updateUser({
+      variables: {
+        where: { id: userId },
+        data: { username, email, password },
+      },
+    })
+      .then((res) => {
+        console.log(res);
+        //form data is not currently refreshing after successful update
+        //loading(false);
+        //probably need to change button text to "Changes Saved" after submit?
+      })
+      .catch(error);
+  };
+
+  const isDesktopSize = useMediaQuery({
+    query: "(min-width: 1170px)",
+  });
+
+  return (
+    <div className="max-w-[1440px] mx-auto">
+      <Header>
+        <Header.Logo />
+        <Header.NavLinks />
+        <Header.Profile />
+      </Header>
+
+      <section className="flex flex-colmax-w-[1440px] h-[900px] bg-gray-background">
+        {isDesktopSize ? <ProfileSidebar /> : <ProfileHorizontalMenu />}
+
+        <section className="mt-9">
+          <Form onSubmit={handleSubmit(onSubmit)}>
+            <Form.TextInput
```

Comment:
All of form must be changed for values to be submittable. Should be that username || email change allows for submission so that way if one or both values change the form can be submitted.